samcli/lib/providers/sam_function_provider.py

```diff
@@ -479,6 +517,11 @@ def _parse_layer_info(
 
             # If the layer is a string, assume it is the arn
             if isinstance(layer, str):
+
+                if search_layer and "arn:aws:lambda" not in layer:
```

Comment:
Partition name `aws` could be different, we need a different way to confirm whether this is ARN or not